src/main/webapp/app/shared/validators/custom-max-length-validator/custom-max-length-validator.directive.ts

```diff
@@ -0,0 +1,21 @@
+import { Directive, Input } from '@angular/core';
+import { FormControl, NG_VALIDATORS, Validator } from '@angular/forms';
+
+/**
+ * Custom max length validator for template-driven forms.
+ */
+@Directive({
+    selector: '[customMaxLength][formControlName],[customMaxLength][formControl],[customMaxLength][ngModel]',
+    providers: [{ provide: NG_VALIDATORS, useExisting: CustomMaxLengthDirective, multi: true }],
+})
+export class CustomMaxLengthDirective implements Validator {
+    @Input({ required: true }) customMaxLength: number;
```

Comment:
I did not get the tests to work with a required input signal, so we use the old version instead